src/llvm_backend.cpp

```diff
@@ -1186,38 +1512,308 @@ gb_internal void lb_finalize_objc_names(lbGenerator *gen, lbProcedure *p) {
+        }
+
+        for (const ObjcMethodData& md : *methods) {
+            lb_handle_objc_find_or_register_selector(p, md.ac.objc_selector);
+        }
+    }
+
+    // Now we can register all referenced selectors
+    for (lbObjCGlobal g = {}; mpsc_dequeue(&gen->objc_selectors, &g); /**/) {
+        lb_register_objc_thing(handled, m, args, class_impls, global_class_map, p, g, "sel_registerName");
+    }
+
+
+    // Emit method wrapper implementations and registration
+    auto wrapper_args     = array_make<Type *>(temporary_allocator(), 2, 8);
+	auto get_context_args = array_make<lbValue>(temporary_allocator(), 1);
+
+
+	PtrMap<Type *, lbObjCGlobal> ivar_map{};
+	map_init(&ivar_map, gen->objc_ivars.count);
+
+	for (lbObjCGlobal g = {}; mpsc_dequeue(&gen->objc_ivars, &g); /**/) {
+		map_set(&ivar_map, g.class_impl_type, g);
+	}
+
+    for (const auto& cd : class_impls) {
+        auto& g = cd.g;
+        Type *class_type = g.class_impl_type;
+    	Type *class_ptr_type = alloc_type_pointer(class_type);
+    	lbValue class_value = cd.class_value;
+
+    	Type *ivar_type = class_type->Named.type_name->TypeName.objc_ivar;
+
+    	Entity *context_provider = class_type->Named.type_name->TypeName.objc_context_provider;
+    	Type *contex_provider_self_ptr_type = nullptr;
+    	Type *contex_provider_self_named_type = nullptr;
+    	bool is_context_provider_ivar = false;
+    	lbValue context_provider_proc_value{};
+
+    	if (context_provider) {
+    		context_provider_proc_value = lb_find_procedure_value_from_entity(m, context_provider);
+
+    		contex_provider_self_ptr_type = base_type(context_provider->type->Proc.params->Tuple.variables[0]->type);
+    		GB_ASSERT(contex_provider_self_ptr_type->kind == Type_Pointer);
+    		contex_provider_self_named_type = base_named_type(type_deref(contex_provider_self_ptr_type));
+
+    		is_context_provider_ivar = ivar_type != nullptr && internal_check_is_assignable_to(contex_provider_self_named_type, ivar_type);
+    	}
+
+
+        Array<ObjcMethodData>* methods = map_get(&m->info->objc_method_implementations, class_type);
+        if (!methods) {
+            continue;
+        }
+
+        for (const ObjcMethodData& md : *methods) {
+            GB_ASSERT( md.proc_entity->kind == Entity_Procedure);
+            Type *method_type = md.proc_entity->type;
+
+            String proc_name = make_string_c("__$objc_method::");
+            proc_name = concatenate_strings(temporary_allocator(), proc_name, g.name);
+            proc_name = concatenate_strings(temporary_allocator(), proc_name, str_lit("::"));
+            proc_name = concatenate_strings( permanent_allocator(), proc_name, md.ac.objc_name);
+
+            wrapper_args.count = 2;
+            wrapper_args[0] = md.ac.objc_is_class_method ? t_objc_Class : class_ptr_type;
+            wrapper_args[1] = t_objc_SEL;
+
+            auto method_param_count  = (isize)method_type->Proc.param_count;
+            i32  method_param_offset = 0;
```

Comment:
Why isn't this just `isize` to begin with?

Reply:
It probably should be. I did it to save space a tiny bit of space in the executable, but perhaps its not worth it, or there may be, in the unlikely case, classes with sizes larger than 4GiB. I'll fix it in the next round.

Reply:
Sorry, I misread here. I thought this referred to the size of the ivar offset global.
Which I changed to native size here: https://github.com/odin-lang/Odin/pull/5064/commits/1505edef011949e10b31335577ea344e5da2b623

Should I revert that or allow sizes above 32-bit address space?

Reply:
Probably ok to leave the ivar offset as native word size. But let me know if you differ.

I fixed the original issue now as well.